torchdynamo/symbolic_convert.py

```diff
@@ -399,6 +399,8 @@ def get_global_source(self, name):
                 source = AttrSource(
                     self.import_source(self.f_globals["__name__"]), name
                 )
+                if name not in self.output.root_globals and name in self.f_globals:
```

Comment:
Funny enough, I think this fix only worked because some completely unrelated code we inline used the global `torch`.  The reference to `torch` was coming from unspecialized integer/float graph inputs and had nothing to do with user code or inlining.

If you modify the inlined function to not use `torch.*` this fix will stop working.

See #1129